synapse/rest/client/v1/room.py

```diff
@@ -291,16 +296,19 @@ async def on_POST(self, request, room_identifier, txn_id=None):
         if RoomID.is_valid(room_identifier):
             room_id = room_identifier
             try:
-                remote_room_hosts = [
-                    x.decode("ascii") for x in request.args[b"server_name"]
-                ]  # type: Optional[List[str]]
-            except Exception:
+                # twisted.web.server.Request.args is incorrectly defined as Optional[Any]
+                args: Dict[bytes, List[bytes]] = request.args  # type: ignore
+
+                remote_room_hosts = parse_strings_from_args(
+                    args, "server_name", required=False
+                )
+            except KeyError:
                 remote_room_hosts = None
```

Comment:
I don't think this can raise a key error anymore? `remote_room_hosts` should be `None` if it isn't found.

Reply:
Oh, duh, yes. Convenient, actually.